kore/src/Kore/Equation/Application.hs

```diff
@@ -112,8 +116,12 @@ attemptEquation sideCondition termLike equation = do
         whileDebugAttemptEquation' . runExceptT $ do
             let Equation{left} = equationRenamed
             (equation', predicate) <- matchAndApplyResults left
-            let Equation{requires} = equation'
-            checkRequires sideCondition predicate requires & whileCheckRequires
+            let Equation{requires, attributes = Attribute.Axiom{simplification}} = equation'
+                eqSrc = Equation.srcLoc equation'
+                onDecidePredicateUnknown = case simplification of
+                    Attribute.NotSimplification -> ErrorInApplication eqSrc
+                    Attribute.IsSimplification _ -> WarnSimplificationEquationInApplication eqSrc
```

Comment:
Can we move this to the `where`? It would make the main body of the function less cluttered.